Parsers/ASimNetworkSession/Parsers/vimNetworkSessionWatchGuardFirewareOS.yaml

```diff
@@ -0,0 +1,196 @@
+      | parse SyslogMessage with * "repeated " EventCount:int " times" *
+      | extend EventCount = iif(isnotempty(EventCount), EventCount, toint(1))
+      | project-away SyslogMessage
+    };
+    let IPParser = (Syslog:(SrcIpAddr:string,DstIpAddr:string)){
+      Syslog
+      | extend temp_SrcMatch = has_any_ipv4_prefix(SrcIpAddr,src_or_any)
+      , temp_DstMatch = has_any_ipv4_prefix(DstIpAddr,dst_or_any)
+      | extend ASimMatchingIpAddr = case(
+        array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-",
+        temp_SrcMatch and temp_DstMatch, "Both",
+        temp_SrcMatch, "SrcIpAddr",
+        temp_DstMatch, "DstIpAddr",
+        "No match"
+      )
+    | where ASimMatchingIpAddr != "No match" 
+    | project-away temp_*
+    };
+    let HostParser = (Syslog:(SrcDomain:string,DstDomain:string)){
+    Syslog
+    | extend temp_SrcMatch = SrcDomain has_any(hostname_has_any)
+      , temp_DstMatch= DstDomain has_any(hostname_has_any)
+    | extend ASimMatchingHostname =case(
+        array_length(hostname_has_any) == 0, "-",
+        temp_SrcMatch and temp_DstMatch, "Both",
+        temp_SrcMatch, "SrcDomain",
+        temp_DstMatch, "DstDomain",
+        "No match"
+      )
+    | where ASimMatchingHostname != "No match" 
+    | project-away temp_*
+    };
+    let AllSyslog = 
+    Syslog
+    | where not(disabled)
+    | where SyslogMessage has_any('msg_id="3000-0148"' 
+        , 'msg_id="3000-0149"' 
+        , 'msg_id="3000-0150"'
+        , 'msg_id="3000-0151"'
+        , 'msg_id="3000-0173"'
+    )
+      and (array_length(ip_any)==0 or has_any_ipv4_prefix(SyslogMessage,ip_any))
+      and (array_length(hostname_has_any)==0 or SyslogMessage has_any(hostname_has_any))
+    | extend DvcAction = extract(@'" (Allow|Deny) ', 1, SyslogMessage)
+    | where (array_length(dvcaction)==0 or DvcAction has_any (dvcaction))
+    | lookup EventLookup on DvcAction
+    | where  (eventresult=="*" or (DvcAction=="Allow" and eventresult=="Success") or (eventresult=="Failure"))
+    | extend NetworkProtocol = extract(@" (tcp|udp|icmp|igmp) ", 1, SyslogMessage)
+    | parse-kv SyslogMessage as (fqdn_src_match:string, fqdn_dst_match:string) with (pair_delimiter=' ', kv_delimiter='=', quote='"')
+    | project-rename DstDomain = fqdn_dst_match, SrcDomain = fqdn_src_match
+    | extend DstDomainType = iif(isnotempty(DstDomain),"FQDN","")
+    | extend SrcDomainType = iif(isnotempty(SrcDomain),"FQDN","")
+    ;
+    let Parse1 = 
+    AllSyslog
+    | where NetworkProtocol !in("icmp","igmp") and SyslogMessage !has "3000-0151"
+    | parse kind=regex flags=U SyslogMessage with * @'" (Allow|Deny) ' RuleName @" \d{2,5} (tcp|udp) \d{2,5} \d{2,5} " SrcIpAddr " " DstIpAddr " " SrcPortNumber:int @" " DstPortNumber:int @" " *
+    | where  (isnull(dstportnumber) or DstPortNumber==dstportnumber)
+    | invoke IPParser()
+    | invoke HostParser()
+    | invoke SyslogParser()
```

Comment:
As a reminder, the reason to have those invokes was to add a project-away in each branch.

Reply:
From AllSyslog, I project only the required fields, then from the Syslog Parser I project away the SyslogMessage